great_expectations_cloud/agent/event_handler.py

```diff
@@ -75,3 +66,32 @@ def handle_event(self, event: Event, id: str) -> ActionResult:
 
 class UnknownEventError(Exception):
     ...
+
+
+class NoVersionImplementationError(Exception):
+    ...
+
+
+def _get_major_version(version: str) -> str:
+    """Get major version as a string. For example, "0.18.0" -> "0"."""
+    parsed: Version = parse_version(version)
+    return str(parsed.major)
+
+
+_GX_MAJOR_VERSION = _get_major_version(gx.__version__)
+
+
+def register_event_action(
+    version: str, event_type: type[Event], action_class: type[AgentAction[Any]]
+) -> None:
+    """Register an event type to an action class."""
+    version_in_map = (
+        _EVENT_ACTION_MAP.get(version) is not None
+    )  # check for None explicitly since version can be empty
+    if version_in_map and (event_type in _EVENT_ACTION_MAP[version]):
```

Comment:
I was trying to raise before we modify `_EVENT_ACTION_MAP`. But if we do raise then it is moot what `_EVENT_ACTION_MAP` is since the program would die because we don't catch the exception raised in `register_event_action()`. I'll simplify.

Reply:
Simplified to use defaultdict and added a test case.